src/content/user-guide/deepsparse-engine/logging.mdx

```diff
@@ -0,0 +1,319 @@
+2. Framework Functions - functions from `torch` or `numpy` (e.g. `torch.mean`, `np.mean`)
+3. Custom Functions - custom user-provided functions with dynamic imports
+
+## Configuration
+
+A YAML file is used to configure both System and Data Logging.
+- **System Logging** is *enabled* if a `logger` is specified
+- **Data Logging** is *disabled* by default. The config allows you to specify what data to log
+
+The YAML file has the following form:
+
+```yaml
+loggers:
+  logger_name1:
+  logger_name2:
+    arg1: 6100                    
+    
+data_logging:
+  # either specify the pipeline stage ... 
+  stage:
+    - func: built_in_fn_name
+      frequency: 1000
+      target_loggers:
+        - logger_name1
+  
+  # ... or a property of the data at a pipeline stage ...
+  stage.target:
+    - func: torch.fn_name
+      frequency: 1000
+      target_loggers:
+        - logger_name2
+
+  # ... or a slice/dict access to a target
+  stage.target[idx]:
+    - func: np.fn_name        
+      frequency: 1000          
+    - func: path/to/custom.py:bar  
+      frequency: 10000
+      # if no target_loggers, defaults to all
+```
+
+Let's break it down piece by piece:
+
+- `loggers` is a dictionary of the form `{logger_name: {config}}` and identifies and provides the arguments for each logger.
+
+- `data_logging` is a dictionary of the form `{stage.target[idx]: [fn_config]}`. 
+For each item, you can provide a list of function configs tha specify which function to apply.
+
+- `stage.target[idx]` specifies if data at a particular `stage` should be logged. If the data at a particular `stage` has
+properties, an optional `target` can be used to unpack the data. If the `target` is a dict or a list, 
+you can access via slicing, indexing, or dict access, for example `stage.target[0:2][0, :-1]['key_3'][5]`.
+
+- `[fn_config]` is list of function configurations of the form `{func: name, frequency: freq, target_loggers: [logger_names]}`. 
+It specifies which the function, the logger, and the frequency of the logging.
+
+<details>
+  <summary>Click for a tangible example for an image classification endpoint.</summary>
+
+```yaml
+loggers:
+  python:         # logs to stdout
+  prometheus:     # logs to prometheus on port 6100
+    port: 6100
+
+endpoints:
+  - task: image_classification
+    route: /image_classification/predict
+    model: zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenet/pruned95_quant-none
+    name: image_classification_pipeline
+    data_logging:
+      pipeline_inputs.images:   # applies to the images (of the form stage.target:)
+        - func: np.shape        # framework function
+          frequency: 1
+          target_loggers:
+            - python          
+
+      pipeline_inputs.images[0]:          # applies to the first image (of the form stage.target[idx]:)
+        - func: mean_pixels_per_channel   # built-in function
```

Comment:
This fails as expected, the function outputs a triple and prometheus does not log triples, only ints/floats